src/Nethermind/Nethermind.Blockchain/Synchronization/BlockDownloader.cs

```diff
@@ -245,6 +260,32 @@ public async Task<long> DownloadBlocks(PeerInfo bestPeer, int newBlocksToSkip, C
                     }
 
                     blocks[indexMapping[i]].Body = body;
+
+                    if (downloadReceipts)
+                    {
+                        if (receipts.Length > i)
```

Comment:
should probably disconnect if receipts are missing

Reply:
Added exceptions